trk/src/Screens/TabScreens/ClimbCreate/Frontend/index.js

```diff
@@ -133,35 +189,92 @@ const ClimbInputData = (props) => {
       ifsc,
       info,
       images: imagesArray,
       setter: setter.uid,
       timestamp: new Date(),
     };
 
-
+    try {
     const { updateClimb } = ClimbsApi();
-    await updateClimb(climbData.id, updatedClimb)
-      .then(async (newClimbId) => {
+    const newClimbId = await updateClimb(climbData.id, updatedClimb);
+    //console.log(newClimbId);
+    if (newClimbId) {
+    //Check if set is a custom set (make), add climb and setter
+        const filteredSets = setItems.filter(set => set.value === 'custom');
+        //console.log(newClimbId);
+
+
+        //Same Logic for deletion
+        if (climbData.set && selectedValue !== climbData.set) {
+          //We need to remove it from its existing set
+          const setObjSnapshot = await SetsApi().getSetByName(climbData.set);
+          if (!setObjSnapshot.empty) {
+            const setObj = setObjSnapshot.docs[0].data();
+            console.log(setObj);
+            const setObjId = setObjSnapshot.docs[0].id;
+            const climbs = setObj.climbs;
+            const setters = setObj.setters;
+            if (climbs.length > 0) {
+              const idx = climbs.indexOf(String(newClimbId));
+              if (idx > -1) {
+                climbs.splice(idx, 1);
+                setters.splice(idx, 1);
+              } else {
+                console.error('Climb has a set, but the set has no record of the climb [ITEMS EXIST]');
+              }
+              await SetsApi().updateSet(setObjId, {climbs: climbs, setters: setters}); //Removing the old climb and setter
+            } else {
+              console.error('Climb has a set, but the set has no record of the climb');
+            }
+          }
+        }
+
+        if (filteredSets.length === 0 && (!climbData || selectedValue !== climbData.set)) { //No need to add if same set
+          //Use the existing setName
+          //console.log('Set already exists!: ', selectedValue);
+          const oldSet = await SetsApi().getSetByName(String(selectedValue).trim());
+          //console.log(oldSet);
+          if (oldSet && oldSet.docs[0]) {
+            const setId = oldSet.docs[0].id;
+            const setData = oldSet.docs[0].data();
+            const newClimbs = [newClimbId].concat((setData.climbs? setData.climbs: [])); //replace with newClimbId
+            const newSetters = [currentUser.uid].concat((setData.setters? setData.setters: []));
+            await SetsApi().updateSet(setId, {climbs: newClimbs, setters: newSetters});
+          } else {
+            console.error('Set Exists But Not Found!');
+          }
+        } else if ((!climbData || selectedValue !== climbData.set)) { //If not a duplicate, then make a custom set
+          const setObj = {
+            archived: false,
+            name: filteredSets[0].name,
+            climbs: [newClimbId], //replace with newClimbId
+            setters: [currentUser.uid]
+          }
+          await SetsApi().addSet(setObj);
+        }
+
```

Comment:
Is the oldSet a set that is archived or no longer live? or does it refer to something different?

Reply:
It is the previous set attached to the Climb. The sets aren't archived (yet), just emptied.